### PR TITLE
Sync files on Podman

### LINKED ISSUE #6335: Re-sync source files when changed

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind feature

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6335 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: pkg/dev/podmandev/reconcile.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	o.deployedPod = pod
+
+	for _, fwPort := range fwPorts {
+		s := fmt.Sprintf("Forwarding from %s:%d -> %d", fwPort.LocalAddress, fwPort.LocalPort, fwPort.ContainerPort)
+		fmt.Fprintf(out, " -  %s", log.SboldColor(color.FgGreen, s))
+	}
+	err = o.stateClient.SetForwardedPorts(fwPorts)
+	if err != nil {
+		return err
+	}
```
Comment (rm3l) [inline pkg/dev/podmandev/reconcile.go:54]: Can we move this to after files have been sync'ed and the `run`/`debug` command has been launched? I found it a bit surprising to display the ports before the application is actually started.

**Review by rm3l — CHANGES_REQUESTED** [review]
Not sure if it was just on my machine, but I just noticed a behavior, which I think could be annoying.
When changes are made, they are correctly re-synced, but `odo dev` keeps detecting changes in the `.odo/devstate.json` file, and keeps restarting the application.
Do you confirm this behavior on your side?

**Comment by feloy** [comment]
> Not sure if it was just on my machine, but I just noticed a behavior, which I think could be annoying. When changes are made, they are correctly re-synced, but `odo dev` keeps detecting changes in the `.odo/devstate.json` file, and keeps restarting the application. Do you confirm this behavior on your side?

Yes you are right, this behaviour happens for every new project where a .gitignore file contains a `.odo+` entry, but not a simple `.odo` entry.

See https://github.com/redhat-developer/odo/issues/6362

**Review by rm3l — APPROVED** [review]

**Comment by feloy** [comment]
/override ci/prow/v4.11-integration-e2e
Infra error
Pass on IBM Cloud

**Pull request merged**